core/runtime/src/main/java/io/quarkiverse/flow/providers/CredentialsProviderSecretManager.java

```diff
@@ -0,0 +1,153 @@
+     */
+    public boolean hasAnyProviders() {
+        return hasAnyProviders;
+    }
+
+    @Override
+    public Map<String, String> secret(String secretName) {
+        CredentialsProvider provider = secretCache.computeIfAbsent(secretName, this::resolveProviderForSecret);
+        Map<String, String> creds = provider.getCredentials(secretName);
+        if (creds == null || creds.isEmpty()) {
+            throw new IllegalStateException("No credentials found for secret '" + secretName
+                    + "' from provider @" + namedOf(provider) + " (" + provider.getClass().getName() + ")");
+        }
+        return creds;
+    }
+
+    private CredentialsProvider resolveProviderForSecret(String secretName) {
+        String per = nullSafeTrim(flowSecrets.credentialsProviderNames().get(secretName));
+        if (per != null) {
+            return selectConfigProvider(per);
+        }
+
+        Optional<String> global = flowSecrets.credentialsProviderName().map(String::trim).filter(s -> !s.isEmpty());
+        if (global.isPresent()) {
+            return selectConfigProvider(global.get());
+        }
+
+        if (singleProvider.isPresent()) {
+            return singleProvider.get();
+        }
+
+        throw new IllegalStateException(
+                "Multiple or no CredentialsProvider beans found and no selection provided. " +
+                        "Set either '" + GLOBAL_KEY + "' or '" + PER_SECRET_KEY.replace("<secret>", secretName) + "'. " +
+                        "Available: " + availableProviders());
+    }
+
+    /**
+     * Centralized config → provider resolution with cache + CDI fallback.
+     */
+    private CredentialsProvider selectConfigProvider(String named) {
+        CredentialsProvider cached = providerBeanCache.get(named);
+        if (cached != null)
+            return cached;
+
+        Instance<CredentialsProvider> handle = providers.select(NamedLiteral.of(named));
+        if (handle.isResolvable() && !handle.isAmbiguous()) {
+            CredentialsProvider p = handle.get();
+            providerBeanCache.putIfAbsent(named, p);
+            return p;
+        }
+        if (handle.isAmbiguous()) {
+            throw new IllegalStateException("Multiple CredentialsProvider beans match @Named='" + named
+                    + "'. Available: " + availableProviders());
+        }
+        throw new IllegalStateException("CredentialsProvider @Named='" + named
+                + "' not found. Available: " + availableProviders());
+    }
+
+    private String availableProviders() {
+        if (!hasAnyProviders)
+            return "<none>";
+        String names = providerBeanCache.isEmpty() ? "<no @Named providers>"
+                : String.join(", ", new TreeSet<>(providerBeanCache.keySet()));
+        String classes = providers.stream()
+                .map(p -> p.getClass().getName())
+                .sorted()
+                .collect(java.util.stream.Collectors.joining(", "));
```

Comment:
```suggestion
                .collect(Collectors.joining(", "));
```
And add the import

Im not really sure the order is consistent here, you are sorting the provider names by alphabetical order and the classes by bean priority, so it might not match, maybe just printing the providerBeanCache will be better